### PR TITLE
Better Config Validation

### LINKED ISSUE #2: Better rule-validation mode
Currently, `tagd` can validate rules and exit, but you still have to specify a Harbor endpoint, username, and password (even if they're blank). These fields should not be required if `--dump-rules` is specified. Furthermore, we should probably make this an explicit `--validate` mode or similar.

### PR DESCRIPTION
**Pull request by nlowe**
Provide an explicit `check` verb for config validation, eliminating "required" parameters that we don't need if we're simply dumping configuration.

Fixes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,28 +49,38 @@ actually delete tags, invoke with `--destructive`.
 ### Usage
 
 ```text
-Usage: Harbor.Tagd [ -h|--help ] [ --version ] --endpoint -u|--userU -p|--passwordP [ -v|--verbosity V ] [ --config-file  ] [ --config-server  ] [ --config-user  ] [ --config-password  ] [ --destructive ] [ --dump-rules ] [ --notify-slack  ]
+Usage: Harbor.Tagd [ -h|--help ] [ --version ] --endpoint -u|--userU [ -p|--password P ] [ --destructive ] [ --notify-slack  ] [ -v|--verbosity V ] [ --config-file  ] [ --config-server  ] [ --config-user  ] [ --config-password  ]
```
Comment (sprutton1) [inline README.md]: `-u|--userU` should be `-u|--user U`

I only point it out because you fixed the password one.

Comment (nlowe) [reply README.md]: This is copied and pasted from `--help`. I'll look into it.

Comment (nlowe) [reply README.md]: See nemec/clipr#46. I don't think this has been released, or if it has, it's in a prerelease

**Comment by nlowe** [comment]
I'm not concerned about the coverage changes. I'll make a separate PR for improved coverage across the board.

**Pull request merged**